src/main/java/gmailgdogra/WriteOutputXlsx.java

```diff
@@ -0,0 +1,90 @@
+package gmailgdogra;
+
+import org.apache.poi.ss.usermodel.*;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class WriteOutputXlsx {
+
+    private static final XSSFWorkbook workbook = new XSSFWorkbook();
+    private static final XSSFSheet sheet = workbook.createSheet("Securitas Daily Report");
+
+    public static boolean write(List<SwipeRecord> outputData) throws IOException {
+
+        addToSheet(OutputRow.of("First Name", "Last Name", "Event Date",
+                "Logical Device"), 0); // title row
+        writeData(outputData);
+        autoSizeColumns();
+        String date = LocalDate.now().format(DateTimeFormatter.ofPattern("dd-MM-yyyy"));
+
+        try (FileOutputStream outputStream = new FileOutputStream("Securitas Report " + date + ".xlsx")) {
+            workbook.write(outputStream);
+        }
+
+        return true;
```

Comment:
Generally, there are a few different approaches how to check a result of an operation: return a boolean; return void on success or throw an exception on failure; return a special class with the result which content will be checked at the call site. You use a mixed approach: return true or throw an exception. It's a bit inconsistent.

Reply:
I'll look into it